contracts/pcv/maker/MakerRouter.sol

```diff
@@ -0,0 +1,196 @@
+        _swapForUsdcAndDai(minDaiAmountOut, ratioUSDC, to);
+    }
+
+    /// @notice Function to swap all FEI balance for both DAI and USDC
+    /// @param to the address the DAI and USDC should be sent to once swapped
+    /// @param ratioUSDC the ratio of the DAI received we would like to swap to USDC - in basis point terms
+    function swapAllFeiForUsdcAndDai(address to, uint256 ratioUSDC)
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        uint256 minDaiAmountOut = _redeemAllBalanceFromFeiPSM(address(this));
+        _swapForUsdcAndDai(minDaiAmountOut, ratioUSDC, to);
+    }
+
+    /// @notice Helper function to redeem DAI from the FEI FEI-DAI PSM
+    /// @param amountFeiIn the amount of FEI to be deposited
+    /// @param minDaiAmountOut the minimum amount of DAI expected to be received
+    /// @param to the address the DAI should be sent to once swapped
+    function _redeemFromFeiPSM(
+        uint256 amountFeiIn,
+        uint256 minDaiAmountOut,
+        address to
+    ) private {
+        fei.safeTransferFrom(msg.sender, address(this), amountFeiIn);
+        feiPSM.redeem(to, amountFeiIn, minDaiAmountOut);
+    }
+
+    /// @notice Helper function to redeem DAI max allowance or balance from the FEI FEI-DAI PSM
+    /// @param to the address the DAI should be sent to once swapped
+    function _redeemAllBalanceFromFeiPSM(address to)
+        private
+        returns (uint256 minDaiAmountOut)
+    {
+        uint256 allowance = fei.allowance(msg.sender, address(this));
+        uint256 amountFeiIn = Math.min(fei.balanceOf(msg.sender), allowance);
+
+        fei.safeTransferFrom(msg.sender, address(this), amountFeiIn);
+        uint256 userBalanceBefore = dai.balanceOf(to);
+
+        // 3 is used as there is a 3 basis points redemption fee
+        minDaiAmountOut =
+            amountFeiIn -
+            (amountFeiIn * 3) /
+            Constants.BASIS_POINTS_GRANULARITY;
+        feiPSM.redeem(to, amountFeiIn, minDaiAmountOut);
+
+        uint256 userBalanceAfter = dai.balanceOf(to);
+
+        require(
+            userBalanceAfter - userBalanceBefore >= minDaiAmountOut,
+            "MakerRouter: Not enough DAI received"
+        );
+    }
+
+    /// @notice Helper function to swap for both DAI and USDC
+    /// @param minDaiAmountOut the minimum amount of DAI expected to be received
+    /// @param to the address the DAI should be sent to once swapped
+    /// @param ratioUSDC the ratio of the DAI received we would like to swap to USDC - in basis point terms
+
+    function _swapForUsdcAndDai(
+        uint256 minDaiAmountOut,
+        uint256 ratioUSDC,
+        address to
+    ) private {
+        uint256 usdcAmount = (minDaiAmountOut * ratioUSDC) /
+            Constants.BASIS_POINTS_GRANULARITY;
+
+        daiPSM.buyGem(to, usdcAmount / USDC_SCALING_FACTOR);
+        dai.safeTransfer(to, minDaiAmountOut - usdcAmount);
+    }
```

Comment:
since this contract is owned by the VOLT Protocol system, we should add a method to sweep ERC20's in case any funds end up in this contract.